scripts/installer/install.sh

```diff
@@ -283,20 +273,39 @@ platform_install_bootloader_entry()
 
 onie_platform=$(onie-sysinfo -p)
 
-# <arch>-<vendor>-<platform>-<revision>
-onl_platform="$(onie-sysinfo -p | tr '_' '-')"
+cd $(dirname $0)
+
+# check that we support the running platform
+if [ ! -d "./machine/${onie_platform}" ]; then
+    echo "ERROR: Unknown or unsupported platform: $onie_platform" >&2
+    echo "This image only supports the following platforms:" >&2
+    platforms="$(ls -1 ./machine)"
+    for plat in $platforms; do
+        echo "    $plat" >&2
+    done
+    exit 1
+fi
+
+# platform.sh may override dummy functions above (e.g., platform_check)
+. ./platform.sh
+
+# setup variables for the running platform
+. ./machine/${onie_platform}/platform.conf
```

Comment:
I would source platform.conf before platform.sh. It should work either way, but variables defined in platform.conf are used by functions defined in platform.sh.

Reply:
Sure, I can switch that.

Reply:
Actually no, since platform.sh was including the platform.conf before, we should first include the flatform.sh, and then the platform.conf. Going from most generic (base installer) to most specific.